source/libs/features.js

```diff
@@ -0,0 +1,153 @@
+import select from 'select-dom';
+import domLoaded from 'dom-loaded';
+import isPromise from 'p-is-promise';
+import OptionsSync from 'webext-options-sync';
+import onAjaxedPages from 'github-injection';
+import onNewComments from './on-new-comments';
+import * as pageDetect from './page-detect';
+import {safeElementReady} from './utils';
+
+const options = new OptionsSync().getAll();
+
+/*
+ *`github-injection` happens even when the user navigates in history
+ * This causes listeners to run on content that has already been updated.
+ * If a feature needs to be disabled when navigating away,
+ * use the regular `github-injection`
+ */
+function safeOnAjaxedPages(callback) {
+	onAjaxedPages(async () => {
+		if (select.exists('has-rgh')) {
+			return;
+		}
+		await domLoaded;
+
+		// Push safeOnAjaxedPages on the next tick so it happens in the correct order
+		// (specifically for addOpenAllNotificationsButton)
+		await Promise.resolve();
+		callback();
+	});
+}
+
+// Rule assumes we don't want to leave it pending:
+// eslint-disable-next-line no-async-promise-executor
+const globalReady = new Promise(async resolve => {
+	await safeElementReady('body');
+
+	if (pageDetect.is500()) {
+		return;
+	}
+
+	if (document.body.classList.contains('logged-out')) {
+		console.warn('%cRefined GitHub%c only works when you’re logged in to GitHub.', 'font-weight: bold', '');
+		return;
+	}
+
+	if (select.exists('html.refined-github')) {
+		console.warn('Refined GitHub has been loaded twice. If you didn’t install the developer version, this may be a bug. Please report it to: https://github.com/sindresorhus/refined-github/issues/565');
+		return;
+	}
+
+	document.documentElement.classList.add('refined-github');
+
+	const {customCSS = ''} = await options;
+	if (customCSS.length > 0) {
+		document.head.append(<style>{customCSS}</style>);
+	}
+
+	resolve();
+});
+
+const run = async (filename, constraints, fn) => {
+	if (!constraints.every(c => c())) {
+		return;
+	}
+	const {disabledFeatures = '', logging = false} = await options;
+	const log = logging ? console.log : () => {};
+
+	if (disabledFeatures.includes(filename)) {
+		log('↩️', 'Skipping', filename);
+		return;
+	}
+	try {
+		// Features can return `false` if they declare themselves as not enabled
```

Comment:
So features that declare themselves as not enabled will go unnoticed in the logging...
Maybe add another log record for those: ⏸

Reply:
The logic is that we only show features on pages that match certain criteria. The base criteria has always been from `pageDetect` but sometimes the feature depends on the page's content (does `element` exist on the page?). I see these two as the same thing. Either we show all the features that try to run or we show only the ones that have run.